Написаны тесты на методы готового класса.

Vezaris opened the pull request:
Написаны тесты на методы готового класса.
Добавлено описание в README.

Vezaris committed:
Написаны тесты на методы готового класса.
Добавлено описание в README.

pomegranate282922 commented on README.md:
```diff
@@ -0,0 +1,8 @@
+ - java 11
+ - junit 4.13.2
+ - mockito 5.5.0
+ - jacoco 0.8.7
```
Отлично что есть ридми

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+
+public class BunTest {
+    private Bun bun;
+    final private String NAME = "булка";
```
Статик нужен для классовой константы 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+public class BunTest {
+    private Bun bun;
+    final private String NAME = "булка";
+    final private float PRICE = 60;
```
И тут

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+    @Test
+    public void getPrice() {
+        assertEquals("Получена неверная стоимость", PRICE, bun.getPrice(), 0.1);
+    }
```
Здесь ждем параметризацию с разными классами эквивалентности 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Test
+    public void getPrice() {
+        assertEquals("Получена неверная стоимость", PRICE, bun.getPrice(), 0.1);
```
Дельту лучше вынести как константу и переиспользовать 
Можно улучшить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,75 @@
+
+    @Test
+    public void getPrice() {
+        final float EXPECTED_PRICE = 130;
```
Не классовые константы именуем начиная со строчной буквы camelCase 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,75 @@
+    final private Bun BUN_SPY = new Bun("Булка", 50);
+    @Spy
+    final private Ingredient INGREDIENT_SPY = new Ingredient(IngredientType.SAUCE, "Соус", 30);
+    @Spy
```
Не уместное применение шпиона. Здесь нужен мок. В ингредиентах аналогично 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/IngredientTest.java:
```diff
@@ -0,0 +1,39 @@
+public class IngredientTest {
+    private Ingredient ingredient;
+    final private String NAME = "Ингредиент";
+    final private float PRICE = 75;
```
Нейминг везде нужно поправить 
Классовые константы - заглавными буквами через нижнее подчеркивание 
Остальные - camelCase

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/IngredientTest.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Test
+    @Parameters ({
+            "SAUCE",
```
Отлично! Уместное применение параметризации

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/IngredientTypeTest.java:
```diff
@@ -0,0 +1,19 @@
+    @Test
+    public void valueFilling() {
+        Assert.assertEquals( IngredientType.valueOf("FILLING"), IngredientType.FILLING);
+    }
```
Здесь тоже просится параметризация 
Можно улучшить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Применена параметризация.
Замечания:
1. Добавить параметризацию в тесты на булочку 
2. Нейминг 
3. Использовать моки вместо шпионов 
4. Остальное по мелочи

Vezaris committed:
Исправлены замечания, а именно:
- для классовых констант добавлен static, для не классовых изменен нейминг
- добавлена параметризация
- дельта для float вынесена в классовую переменную
- шпионы заменены на моки

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/BunTest.java:
```diff
@@ -0,0 +1,46 @@
+  public static Object[][] getDataEntry() {
+    return new Object[][]{
+        {"black bun", 100},
+        {"white bun", 200},
```
Ну это не так чтобы разные классы эквивалентности 
Имелось в виду: для строк - кириллица, латиница, пустое, null и тп 
Для чисел - аналогично 
Можно улучшить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/IngredientTest.java:
```diff
@@ -0,0 +1,51 @@
+  private final IngredientType ingredientType;
+  private final String name;
+  private final float price;
+  private static final float DELTA = 0.1F;
```
Можно было одну дельту на проект сделать

pomegranate282922 reviewed (approved):
Отлично, зачет!
